src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1335,13 +1496,11 @@ public void setSoLinger(boolean on, int linger) throws SocketException {
         }
 
         public void setSoTimeout(int timeout) throws SocketException {
-            if (logger.isLoggable(Level.FINER))
-                logger.finer(toString() + " Ignoring setSoTimeout");
+            tdsChannel.tcpSocket.setSoTimeout(timeout);
```

Comment:
This looked like a bug to me. It was resulting in hangs during testing when the connection was encrypted since the new disconnect detection sets this to 1ms to check for a timeout. I suspect this may be cause other hang issues.